### PR TITLE
Fix for using external camp

### PR DESCRIPTION
**Pull request by adayton1**
This pull request adds some fixes for using external camp.

### CODE REVIEW

File: src/umpire/tpl/CMakeLists.txt [hunk 1]
```diff
@@ -114,7 +114,6 @@ if (NOT TARGET camp)
     blt_register_library(
       NAME camp
       DEPENDS_ON cuda_runtime
-      LIBRARIES camp
```
Comment (davidbeckingsale) [inline src/umpire/tpl/CMakeLists.txt:117]: @adayton1 - why is this removed?

Comment (adayton1) [reply src/umpire/tpl/CMakeLists.txt:117]: Doesn't this try to add -lcamp to the link line?

Comment (adayton1) [reply src/umpire/tpl/CMakeLists.txt:117]: @white238 @gzagaris camp is a header only library, so should `LIBRARIES camp` be left out of the call to `blt_register_library`?

Comment (white238) [reply src/umpire/tpl/CMakeLists.txt:117]: So it does a bunch of possible things:

1) If it is a BLT object library (which is done by blt_add_library(...OBJECT TRUE...)) it circumvents the normal CMake object libraries that have a lot of side effects that are very unhelpful.

2) If it is a BLT registered library, it adds all the known information from the previous blt_register_library

(the one its probably doing) 3) The normal case calls target_link_library() (which is a normal CMake function), which if its a CMake target, inherits information following the CMake normal rules, or if it is not a CMake target, it blindly adds -lcamp to the link line in hopes that it will find it in the system or an already added -L directory.

If you need more clarification I can look into it further.

Comment (davidbeckingsale) [reply src/umpire/tpl/CMakeLists.txt:117]: @white238 - camp provides a proper CMake target regardless of whether it's "external" or not. The only extra requirement is the `CAMP_HAVE_CUDA` define to make sure it enables stuff that's guarded in the headers.

What's the best way to do this?

**Comment by davidbeckingsale** [comment]
@adayton1 - can you give this a try?

**Comment by davidbeckingsale** [comment]
@adayton1 - just wanted to confirm you are happy with this after our email discussion? I will go ahead and update the installed CMake config in another PR to handle the CHAI issue.

**Comment by davidbeckingsale** [comment]
LGTM

**Comment by adayton1** [comment]
This looks good to me now. I tried to review it, but github is smart enough to not let me review my own pull request :)

**Comment by adayton1** [comment]
Another thought occurred to me: Could camp configure a header so that we don't have to add defines? That would also guarantee consistency across projects.

**Comment by davidbeckingsale** [comment]
> Another thought occurred to me: Could camp configure a header so that we don't have to add defines? That would also guarantee consistency across projects.

I replied to the issue you opened on camp.
